taxonomy/openai/client.py

```
@@ -9,23 +9,35 @@
 log = logging.getLogger(__name__)
 
 
-def chat_completion(prompt):
+def chat_completion(prompt, system_message):
     """
-    Pass message list to chat endpoint, as defined by the CHAT_COMPLETION_API setting.
+    Pass message list to chat endpoint, as defined by the XPERT_AI_API_V2 setting.
     Arguments:
         prompt (str): chatGPT prompt
+        system_message (str): system message to be used in the chat
     """
-    completion_endpoint = settings.CHAT_COMPLETION_API
-    completion_endpoint_key = settings.CHAT_COMPLETION_API_KEY
-    headers = {'Content-Type': 'application/json', 'x-api-key': completion_endpoint_key}
+    completion_endpoint = settings.XPERT_AI_API_V2
+    headers = {'Content-Type': 'application/json'}
     connect_timeout = getattr(settings, 'CHAT_COMPLETION_API_CONNECT_TIMEOUT', 1)
     read_timeout = getattr(settings, 'CHAT_COMPLETION_API_READ_TIMEOUT', 15)
-    body = {'message_list': [{'role': 'assistant', 'content': prompt},]}
+    body = {
+        'messages': [{'role': 'user', 'content': prompt},],
+        'client_id': settings.XPERT_AI_CLIENT_ID,
+        'system_message': system_message,
+    }
     response = requests.post(
         completion_endpoint,
         headers=headers,
         data=json.dumps(body),
         timeout=(connect_timeout, read_timeout)
     )
-    chat = response.json().get('content')
-    return chat
+
+    if response.status_code != 200:
+        log.error(
+            'Error in chat completion API: %s, %s',
+            response.status_code,
+            response.text
+        )
+        return ''
```

Instead of returning an empty string, I think we should re-raise the error.